docs/user-manual/plugin-system.md

```diff
@@ -0,0 +1,255 @@
+
+# Writing a plugin
+
+A plugin is a simple c++ class that is declared in a compiled component, e.g shared library or executable. No inheritance is needed to declare a plugin. The macro **DQM_PLUGIN_DECL()** takes care of everything for you. As always, an example is better than a long documentation :
+
+Header file: 
+```cpp
+#include <iostream>
+
+class MyPlugin {
+public:
+  MyPlugin() {};
+  void helloWorld(); 
+};
+```
+
+Source file:
+```cpp
+#include <MyPlugin.h>
+#include <dqm4hep/PluginManager.h>
+
+MyPlugin::helloWorld() { 
+  std::cout << "Hello world !" << std::endl; 
+}
+
+// This is where the magic happens !
+DQM_PLUGIN_DECL( MyPlugin, "MyPlugin" );
+```
+
+Most of the code is just a simple c++ class that does nothing else than saying "Hello world !". In the source file, the *PluginManager.h* include is needed to declare your plugin in the registry. The last line of the source actually does the plugin declaration. The first argument is the class name itself and the second, the plugin name that can be different from the class name. 
+
+<div class="warning-msg">
+  <i class="fa fa-warning"></i>
+  The registration macro performs operations that require a simple class name, for example without namespace specifier in the argument. You can use <span style="font-style: italic">'typedef'</span> or <span style="font-style: italic">'using'</span> keywords to fix the problem.
+</div>
+
+The source file can be compiled in a shared library or within a binary and loaded at runtime (see next section).
+
+# The PluginManager API
+
+This section explains how the declared plugins are loaded and accessed from the **PluginManager** class.
+
+First of all, we need to load the shared libraries in which the plugin are implemented. This can be done in different ways :
+
+```cpp
+#include <dqm4hep/PluginManager.h>
+
+using namespace dqm4hep::core;
+
+int main() {
+  
+  // get the pointer on the plugin manager
+  auto mgr = PluginManager::instance();
+  
+  // load a single library
+  THROW_RESULT_IF( STATUS_CODE_SUCCESS, != , mgr->loadLibrary( "libMyPlugin.so" ) );
+  
+  // load a set of libraries
+  THROW_RESULT_IF( STATUS_CODE_SUCCESS, != , mgr->loadLibraries( "libMyPlugin.so:libMyPlugin2.so:libSuperman.so" ) );
+  
+  // load libraries from the environment variable DQM4HEP_PLUGIN_DLL
+  THROW_RESULT_IF( STATUS_CODE_SUCCESS, != , mgr->loadLibraries() );
+  
+  return 0;
+}
+```
+
+<div class="info-msg">
+  <i class="fa fa-info-circle"></i>
+  Most of the binaries provided the DQM4hep packages load the shared libraries using the third option, with the environment variable <span style="font-style: italic">DQM4HEP_PLUGIN_DLL</span>, at the beginning of the main. 
+</div>
+
+Before running a DQM4hep program it is usually a good idea to export <span style="font-style: italic">DQM4HEP_PLUGIN_DLL</span> with your plugin libraries inside :
+
+```shell
+export DQM4HEP_PLUGIN_DLL=libSpiderman.so:libThor.so:libIronman.so
```

Comment:
Why are you mixing DC and Marvel characters?

Reply:
Super heroes are super heroes. They save the world, that's all ...